src/lib/util/utils.ts

```diff
@@ -447,17 +463,28 @@ exports.getJsonContentType = function getJsonContentType(consumesOrProduces) {
  * @param {string} str - The input string to be verified.
  * @returns {boolean} result - true if str is url encoded; false otherwise.
  */
-exports.isUrlEncoded = function isUrlEncoded(str) {
+export function isUrlEncoded(str: string) {
   str = str || '';
   return str !== decodeURIComponent(str);
 };
 
+export type Model = {
+  type?: string|ReadonlyArray<string>
+  readonly properties?: {
+    [key: string]: Model|undefined
+  }
+  readonly required?: ReadonlyArray<unknown>
+  additionalProperties?: {
+    type: ReadonlyArray<string>
+  }
+}
+
 /**
  * Determines whether the given model is a pure (free-form) object candidate (i.e. equivalent of the C# Object type).
  * @param {object} model - The model to be verified
  * @returns {boolean} result - true if model is a pure object; false otherwise.
  */
-exports.isPureObject = function isPureObject(model) {
+export function isPureObject(model: Model) {
```

Comment:
Not for this PR, but given the code that's been copied from oav, we should extract out and share, seems like this tool needs a bunch of restructuring.

Reply:
I'm considering to create a package for OpenAPI 2.0 types and parsing.